### PR TITLE
Backport 2.12 to 0.7.x branch

### PR DESCRIPTION
**Pull request by johnynek**
This brings e5afee8da3681829a0ed0fb8edd9549488f4efaf to the 0.7.x (Kryo 2.21) branch.

### CODE REVIEW

**Comment by johnynek** [comment]
@sritchie @isnotinvain review?

File: README.md [hunk 1]
```diff
@@ -157,9 +162,21 @@ The latest ScalaDocs are hosted on Chill's [Github Project Page](http://twitter.
...
+The TL;DR is send us a pull request, iterate on the feedback + discussion, and get a +1 from a [Committer](COMMITTERS.md) in order to get your PR accepted.
+
+The current list of active committers (who can +1 a pull request) can be found here: [Committers](COMMITTERS.md)
+
+A list of contributors to the project can be found here: [Contributors](https://github.com/twitter/chill/graphs/contributors)
+
 ## Maven
 
-Chill modules are available on Maven Central. The current groupid and version for all modules is, respectively, `"com.twitter"` and  `0.7.4` and each scala project is published for `2.10` and `2.11`. Search [search.maven.org](http://search.maven.org/#search%7Cga%7C1%7Cchill) when in doubt.
+Chill modules are available on Maven Central. The current groupid and version for all modules is, respectively, `"com.twitter"` and  `0.8.0` and each scala project is published for `2.10`, `2.11` and `2.12`. Search [search.maven.org](http://search.maven.org/#search%7Cga%7C1%7Cchill) when in doubt.
```
Comment (sritchie-stripe) [inline README.md:179]: do we care about the README on each of these branches reflecting the 0.7.x, etc and proper kryo version?

File: build.sbt [hunk 2]
```diff
@@ -1,23 +1,14 @@
-import ReleaseTransformations._
```
Comment (sritchie-stripe) [inline build.sbt:1]: Are we also going to have to backport the doge release business? I think so.

File: chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala [hunk 3]
```diff
@@ -41,6 +43,14 @@ class KryoBase extends Kryo {
   def isFn(klass: Class[_]): Boolean =
     functions.find { _.isAssignableFrom(klass) }.isDefined
 
+  def isJavaLambda(klass: Class[_]): Boolean =
+    Java8ClosureRegistrar.areOnJava8 && klass.getName().indexOf('/') >= 0
+
+  override def getRegistration(klass: Class[_]) =
+    if (isJavaLambda(klass)) {
+      getClassResolver.getRegistration(classOf[ClosureSerializer.Closure])
```
Comment (sritchie-stripe) [inline chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala:51]: hmm, odd that this doesn't have coverage even though without it the tests failed.

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,9 +1,9 @@
-resolvers += Resolver.url("sbt-plugin-releases", new URL("http://scalasbt.artifactoryonline.com/scalasbt/sbt-plugin-releases/"))(Resolver.ivyStylePatterns)
-
 addSbtPlugin("com.github.gseitz" % "sbt-release"     % "1.0.0")
```
Comment (sritchie-stripe) [inline project/plugins.sbt]: we'll need to bump this as well - that'll get covered by the cherry-pick, I guess.

**Review by sritchie-stripe — CHANGES_REQUESTED** [review]

**Comment by sritchie-stripe** [comment]
Huh, on 2.12:

```
[info] - should clean actual closures *** FAILED ***
[info]   java.lang.IllegalArgumentException:
[info]   at com.esotericsoftware.reflectasm.shaded.org.objectweb.asm.ClassReader.<init>(Unknown Source)
[info]   at com.esotericsoftware.reflectasm.shaded.org.objectweb.asm.ClassReader.<init>(Unknown Source)
[info]   at com.esotericsoftware.reflectasm.shaded.org.objectweb.asm.ClassReader.<init>(Unknown Source)
[info]   at com.twitter.chill.ClosureCleaner$.getClassReader(ClosureCleaner.scala:62)
[info]   at com.twitter.chill.ClosureCleaner$.getInnerClassesFn(ClosureCleaner.scala:112)
[info]   at com.twitter.chill.ClosureCleaner$.$anonfun$innerClassesOf$1(ClosureCleaner.scala:125)
[info]   at com.twitter.chill.ClosureCleaner$$$Lambda$1983/216374448.apply(Unknown Source)
[info]   at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:79)
[info]   at com.twitter.chill.ClosureCleaner$.innerClassesOf(ClosureCleaner.scala:125)
[info]   at com.twitter.chill.ClosureCleaner$.getAccessedFields(ClosureCleaner.scala:133)
[info]   ...
```

File: project/plugins.sbt (hunk 4 above)
Comment (johnynek) [reply project/plugins.sbt]: huh, I guess we overlooked this.

File: chill-akka/src/main/scala/com/twitter/chill/akka/ActorRefSerializer.scala [hunk 5]
```diff
@@ -45,8 +41,8 @@ class ActorRefSerializer(system: ExtendedActorSystem) extends Serializer[ActorRe
   }
 
   override def read(kryo: Kryo, input: Input, typ: Class[ActorRef]): ActorRef = {
-    val path = input.readString()
-    system.actorFor(path)
+    val path = ActorPath.fromString(input.readString())
```
Comment (sritchie-stripe) [inline chill-akka/src/main/scala/com/twitter/chill/akka/ActorRefSerializer.scala:44]: Oh! Nice, I didn't realize that I DID merge these changes and that they're backwards compatible. Nice.

**Comment by sritchie-stripe** [comment]
I think there's a race on the build, kick it again but 👍  on the code

**Pull request merged**